Salwador/Services/FileManagerService.swift

```diff
@@ -29,4 +29,14 @@ class FileManagerService {
 
         return nil
     }
+
+    func getApiKey() -> String {
+        let data = readFile(fileName: "APIKey", fileType: "json")
+        do {
+            let json = try JSONDecoder().decode(APIJSONModel.self, from: data!)
+            return json.apikey
+        } catch {
+            return ""
+        }
+    }
 }
```

Comment:
idea: i would suggest something that would allow you to handle underlying errors upstream. 

```swift
public enum FileManagerError: Error {
    case PathNotFoundInBundle
    case URL
}

struct FileManagerService {
    func readFile(fileName: String, fileType: String) throws -> Data {
        guard let filePath = Bundle.main.path(forResource: fileName, ofType: fileType) else {
            throw FileManagerError.PathNotFoundInBundle
        }

        guard let fileURL = URL(string: filePath) else {
            throw FileManagerError.URL
        }

        return try Data(contentsOf: fileURL)
    }
}
```

And i would also refactor this accordingly.

```swift
public enum APIKeyError: Error {
    case APIKeyNotFound
}

struct APIKeyProvider {
    let fileManager = FileManagerService()

    func getAPIKey() throws -> String {
        let data = try fileManager.readFile(fileName: "APIKey", fileType: "json")
        let json = try JSONDecoder().decode(APIJSONModel.self, from: data)
        return json.apikey
    }
}
```

Reply:
https://sarunw.com/posts/different-ways-to-catch-throwing-errors-in-swift/